src/requirementslib/models/requirements.py

```diff
@@ -850,23 +908,99 @@ def as_pipfile(self):
             conflicts = [k for k in (conflicting_keys[1:],) if k in base_dict]
             for k in conflicts:
                 base_dict.pop(k)
-        if "hashes" in base_dict and len(base_dict["hashes"]) == 1:
-            base_dict["hash"] = base_dict.pop("hashes")[0]
+        if "hashes" in base_dict:
+            _hashes = base_dict.pop("hashes")
+            hashes = []
+            for _hash in _hashes:
+                try:
+                    hashes.append(_hash.as_line())
+                except AttributeError:
+                    hashes.append(_hash)
+            base_dict["hashes"] = sorted(hashes)
         if len(base_dict.keys()) == 1 and "version" in base_dict:
             base_dict = base_dict.get("version")
         return {name: base_dict}
 
+    def as_ireq(self):
+        ireq_line = self.req.req.line
+        if ireq_line.startswith("-e "):
+            ireq_line = ireq_line[len("-e "):]
+            ireq = InstallRequirement.from_editable(ireq_line)
+        else:
+            ireq = InstallRequirement.from_line(ireq_line)
+        ireq.extras = tuple(self.extras) if self.extras else ()
+        ireq.markers = self.markers
```

Comment:
So you can't manipulate anything on an ireq directly.  Remember when I had all that logic to ensure that there was a proper `packaging.requirements.Requirement` object stored in `ireq.req`? `ireq.markers` comes from `ireq.req.marker`, same with `ireq.extras` :D

Reply:
I wonder if it’s possible to use `as_line` for this instea.